### DISCUSSION TITLE
Default canvas-click pan and zoom settings?

### DISCUSSION THREAD

**Discussion by DevinSmithWork** [openseadragon/openseadragon#2480] [Q&A] [answered]
Hi, I'm using an HTML button as an overlay, and need to have two separate workflows when a user (1) clicks on the button overlaying the canvas, and (2) clicks elsewhere on the canvas.

Presently, I'm using a handler for the canvas-click event and disabling the default action:
```
viewer.addHandler('canvas-click', function(event) {
        event.preventDefaultAction = true
        ...
```

However, I would actually like to replicate the default pan/zoom action if (2) the user doesn't click the button. I'm wondering if these pan/zoom calculations are somewhere in the code I could reference?

**Comment by iangilman** [comment, marked as answer]
So you want the normal behavior everywhere except where the button is? This codepen might be helpful: 

https://codepen.io/msalsbery/pen/dyOeXqO

I think you're going to be better off leaving the existing behavior alone rather than trying to replicate it. That said, here's where panning is implemented:

https://github.com/openseadragon/openseadragon/blob/7dfc60a97fa2ce88213d7a1228749fd7f444aa65/src/viewer.js#L3123

**Reply by DevinSmithWork** [reply]
This codepen looks like what I need. I'll take a look over the weekend and report back. Thank you!!

**Reply by msalsbery** [reply]
@DevinSmithWork  Did you get that working the way you needed?

**Reply by DevinSmithWork** [reply]
Yup! Ian's codepen was right on the money.

**Reply by msalsbery** [reply]
@DevinSmithWork  Awesome thanks!
